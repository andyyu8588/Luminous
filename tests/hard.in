// For compiler/VM testing purpose

a = "string";
b = "number";
c = a + b + "null" + "boolean";
d = not(not(1 * 0 - 22 * -1)) equals true;
{
  print(a equals b);
  a = null;
  print(a equals b);
  print(a);
  b = null;
  print(a);
  print(not b);
  print(a equals b);
  {
    print(a equals b);
    b = 4.13;
    print(a equals b);
    c = 4.13;
    print(c > b);
    {
      d = c;
      e = d;
      print(e equals d);
      f = "nice";
      g = "ok";
      h = "testtest123123";
      {
        f = g;
        g = h;
        h = f + g;
        print(h);
      }
    }
  }
}
print(a);
print(b);
print(c);
print(d);