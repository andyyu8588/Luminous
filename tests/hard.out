false
false
null
null
true
true
true
false
false
true
oktesttest123123
string
number
stringnumbernullboolean
true
